src/os/os.c

```diff
@@ -856,19 +884,19 @@ void main_thread(void *arg)
 
     /* setup the monitoring entries for the timer and idle tasks */
 #if configUSE_TIMERS
-    taskList.next = malloc(sizeof(TaskList)*2);
-    taskList.next->task = xTimerGetTimerDaemonTaskHandle();
-    taskList.next->unused = uxTaskGetStackHighWaterMark(taskList.next->task);
-    taskList.next->next = taskList.next + 1;
-    taskList.next->next->task = xTaskGetIdleTaskHandle();
-    taskList.next->next->unused = uxTaskGetStackHighWaterMark(taskList.next->next->task);
-    taskList.next->next->next = NULL;
+    TaskList *task_new = malloc(sizeof(TaskList) * 2);
+    task_new->task = xTimerGetTimerDaemonTaskHandle();
+    task_new->name = (char*)pcTaskGetTaskName(task_new->task);
+    task_new->unused = uxTaskGetStackHighWaterMark(task_new->task);
+    add_thread_to_task_list(task_new);
+    ++task_new;
 #else
-    taskList.next = malloc(sizeof(TaskList));
-    taskList.next->task = xTaskGetIdleTaskHandle();
-    taskList.next->unused = uxTaskGetStackHighWaterMark(taskList.next->task);
-    taskList.next->next = NULL;
+    TaskList *task_new = malloc(sizeof(TaskList));
 #endif
+    task_new->task = xTaskGetIdleTaskHandle();
+    task_new->name = (char*)pcTaskGetTaskName(task_new->task);
```

Comment:
There is a bit of code here that's repeated three times, maybe four. Could this be moved inside the helper function maybe?